Cron/Ecommerce.php

```diff
@@ -128,8 +135,8 @@ public function execute()
                     $this->_apiResult->processResponses($storeId, true, $mailchimpStoreId);
                     $batchId = $this->_processStore($storeId, $mailchimpStoreId, $listId);
                     if ($batchId) {
-                        $connection->update($tableName, ['batch_id' => $batchId, 'mailchimp_sync_modified' => 0, 'mailchimp_sync_delta' => $this->_helper->getGmtDate()], "batch_id is null and mailchimp_store_id = '$mailchimpStoreId'");
-                        $connection->update($tableName, ['batch_id' => $batchId, 'mailchimp_sync_modified' => 0, 'mailchimp_sync_delta' => $this->_helper->getGmtDate()], "batch_id is null and mailchimp_store_id = '$listId'");
+                        $connection->update($tableName, ['batch_id' => $batchId, 'mailchimp_sync_modified' => 0, 'mailchimp_sync_delta' => $this->_helper->getGmtDate()], "batch_id is null and mailchimp_store_id = '".$this->escaper->escapeHtml($mailchimpStoreId)."'");
```

Comment:
Escaping html isn't proper fix.

Looks like this code should be like this:

```suggestion
                        $connection->update(
                            $tableName,
                            ['batch_id' => $batchId, 'mailchimp_sync_modified' => 0, 'mailchimp_sync_delta' => $this->_helper->getGmtDate()],
                            ["batch_id is null and mailchimp_store_id = ?" => $mailchimpStoreId]
                        );
```

I took this example from 
https://github.com/magento/magento2/blob/b2ce2a37d921b5ad88fc38663fc0ff3dd6c582d1/app/code/Magento/Reports/Model/ResourceModel/Event.php#L67-L71